File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

int get_cents(void)
{
    // Prompt user for number of cents (positive) and return the same number
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 1);

    return cents;
}

int calculate_quarters(int cents)
{
    // Calculate and return the number of quarters based on the cents owed
    for (int q = 0; cents >= 25; q++);
    {
        q = (cents/25);
        
    }
}

int calculate_dimes(int cents)
{
    // Calculate and return the number of dimes based on cents owed
    int d;
    do
    {
        d = (cents/10);
    }
    while (cents < 10);
    return d;
}

int calculate_nickels(int cents)
{
    // Calculate and return the number of nikels based on cents owed
    int n;
    do
    {
        n = (cents/5);
    }
    while (cents < 5);
    return n;
}

int calculate_pennies(int cents)
{
    // Calculate and return the number of pennies based on the cents owed
    int p;
    do
    {
        p = (cents/1);
    }
    while (cents < 1);
    return p;
}

File: python/sentimental-cash/cash.py
from cs50 import get_float

def main():
    # Prompt user for dollars owed
    dollars = get_dollars()

    # Calculate the number of quarters to give
    quarter_count = quarters(dollars)
    # Update the dollars after subtracting the ones used in quarters
    dollars = dollars - (quarters * 0.25)

    dime_count = dimes(dollars)
    # Update dollars
    dollars = dollars - (dimes * 0.10)

    # Calculate number of nickels
    nickel_count = nickels(dollars)
    # Update dollars
    dollars = dollars - (nickels * 0.5)

    # Calculate pennies
    penny_count = pennies(dollars)
    # Update dollars
    dollars = dollars - (pennies * 0.10)

    # Count the number of coins
    coins = quarter_count + dime_count + nickel_count + penny_count

    print(coins)


def get_dollars():
    while True:
        try:
            user_input = get_float("Change owed: ")
            if user_input > 0:
                return user_input
            else:
                print("Enter a positive value")
        except ValueError:
            print("Enter a valid digit")

def quarters(dollars):
    # Initializing a variable to count quarters
    q = 0
    while dollars >= 0.25:
        q += 1
        dollars -= 0.25
    return q


def dimes(dollars):
    d = 0
    while dollars >= 0.10:
        d += 1
        dollars -= 0.10
    return d


def nickels(dollars):
    n = 0
    while dollars >= 0.5:
        n += 1
        dollars -= 0.5
    return n


def pennies(dollars):
    p = 0
    while dollars >= 0.1:
        p += 1
        dollars -= 0.1
    return p


main()
